advanced/security.md

```diff
@@ -2,46 +2,141 @@
+| `img-src 'self' _.tinymce.com _.tiny.cloud data: blob:;`       | Images within the editor are sometimes base64 encoded, blob URLs, or proxied through the {{site.cloudname}} service.         |
+| `style-src 'self' 'unsafe-inline' _.tinymce.com _.tiny.cloud;` | Styles are used for inline formatting (such as underline, font colors, etc.) and the positioning of user interface elements. |
+| `font-src 'self' _.tinymce.com _.tiny.cloud;`                  | Fonts are used for icons in the UI and is loaded from external files.                                                        |
 
 > **Important**: These directives will prevent all content loading from external sources.
 > To allow content from specific sources, add the source domains to the relevant directives. For example, to allow YouTube videos:
->  ```html
->  media-src 'self' *.youtube.com;
->  ```
->  To allow content from any source, use the (&#42;) wildcard. Allowing all sources (using &#42;) negates the security policy for the source type. For example:
->  ```html
->  media-src *;
->  ```
+>
+> ```html
+> media-src 'self' *.youtube.com;
+> ```
+>
+> To allow content from any source, use the (\*) wildcard. Allowing all sources (using \*) negates the security policy for the source type. For example:
+>
+> ```html
+> media-src *;
+> ```
+>
 > For information on Content Security Policies, see: [MDN Web Docs - Content Security Policy (CSP)](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP).
 
 When using the {{site.cloudname}}, use this CSP header :
 
 ```html
-<meta http-equiv="Content-Security-Policy" content="default-src 'none'; script-src 'self' *.tinymce.com *.tiny.cloud; connect-src 'self' *.tinymce.com *.tiny.cloud blob:; img-src 'self' *.tinymce.com *.tiny.cloud data: blob:; style-src 'self' 'unsafe-inline' *.tinymce.com *.tiny.cloud; font-src 'self' *.tinymce.com *.tiny.cloud;" />
+<meta
+  http-equiv="Content-Security-Policy"
+  content="default-src 'none'; script-src 'self' *.tinymce.com *.tiny.cloud; connect-src 'self' *.tinymce.com *.tiny.cloud blob:; img-src 'self' *.tinymce.com *.tiny.cloud data: blob:; style-src 'self' 'unsafe-inline' *.tinymce.com *.tiny.cloud; font-src 'self' *.tinymce.com *.tiny.cloud;"
+/>
 ```
 
-When self-hosting {{site.productname}} from a local domain, use this CSP header (excludes the &#42;.tiny.cloud domain):
+When self-hosting {{site.productname}} from a local domain, use this CSP header (excludes the \*.tiny.cloud domain):
 
 ```html
-<meta http-equiv="Content-Security-Policy" content="default-src 'none'; script-src 'self'; connect-src 'self' blob:; img-src 'self' data: blob:; style-src 'self' 'unsafe-inline'; font-src 'self';" />
+<meta
+  http-equiv="Content-Security-Policy"
+  content="default-src 'none'; script-src 'self'; connect-src 'self' blob:; img-src 'self' data: blob:; style-src 'self' 'unsafe-inline'; font-src 'self';"
+/>
 ```
+
+## Potential security risks and mitigation measures
```

Comment:
This whole section really feels out of place to be honest, as most of it is completely outside the scope of TinyMCE and really depends on how or where it's being integrated. As an example, if I was using TinyMCE in wordpress this would be rather confusing as they aren't implementing the things that would be affected themselves.

I'd suggest scrapping everything that doesn't impact TinyMCE directly and just include a heading that links to the OWASP 10 as things to be aware of.

Reply:
As the first paragraph may indicate, I added this section as general information about potential risks related to user inputs, and to highlight that no editor will necessarily protect users/servers from attack.

Reply:
I still find it confusing having it like this though. Maybe we can break it out into a "Other security issues to be aware of outside TinyMCE" or similar?

Reply:
renamed "General security risks for user input elements"